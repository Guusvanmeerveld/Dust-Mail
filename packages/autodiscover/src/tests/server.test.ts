import fetchServerWithThunderbird from "../thunderbird";
// import fetchServerWithAutodiscover from "../autodiscover";

const username = "mail@guusvanmeerveld.dev";
const server = username.split("@").pop() as string;

describe("Validate autodiscover checker", () => {
	// test("Test autodiscover functionality", async () => {
	// 	await fetchServerWithAutodiscover(server)
	// 		.catch((error: Error) =>
	// 			expect(error.message).toBe("Email adress is not valid")
	// 		)
	// 		.then((data) => expect(data).toBeUndefined());
	// });

	test("Test Thunderbird autoconfig discovery functionality", async () => {
		await fetchServerWithThunderbird(server, username).then((data) => {
			expect(data).toMatchSnapshot();
		});
	});
});
