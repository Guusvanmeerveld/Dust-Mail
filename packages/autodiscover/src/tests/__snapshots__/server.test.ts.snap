// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate autodiscover checker Test Thunderbird autoconfig discovery functionality 1`] = `
Array [
  Object {
    "port": 993,
    "security": "TLS",
    "server": "imap.guusvanmeerveld.dev",
    "type": "imap",
  },
  Object {
    "port": 587,
    "security": "STARTTLS",
    "server": "smtp.guusvanmeerveld.dev",
    "type": "smtp",
  },
]
`;
